spec/models/searchgov_url_spec.rb

```diff
@@ -890,6 +890,75 @@
       end
     end
 
+    context 'when url has different canonical_url' do
+      let(:canonical_url) { 'https://www.agency.gov/new.html' }
+
+      before do
+        expect(I14yDocument).not_to receive(:create).
```

Comment:
We should avoid using `expect` in before blocks, and this expectation actually seems unneeded/unrelated.

Also, let's set up a spy here so that we can use the `have_received` syntax in the expectation below.

So, here in the before:

`allow(described_class).to receive(:create)`